### ISSUE TITLE
SendTradeOfferWithLink

### ISSUE THREAD

**Issue by AlekseySkynox** [FatherFoxxy/CSharpTradeOffers#17]
Hello. Do not work trade (SendTradeOfferWithLink). 
Method "Retry Request Processor" returns an error 400 or 500. May be error in my code.
An excerpt of code:

```

var AccountIdOther = uint.Parse("305044***");
                var partnerId = IdConversions.AccountIdToUlong(AccountIdOther);
var token = "jdgHx***";

var myInvHandler = new InventoryHandler(_account.SteamId);
uint[] appids = { AppId };
myInvHandler.RefreshInventories(appids);

TradeOffer offer = new TradeOffer();
RgInventoryItem item = myInvHandler.Inventories[AppId].Items.First().Value.Items.First();
offer.Me.Assets.Add(item.ToCEconAsset(AppId));

OfferHandler.SendTradeOfferWithLink(partnerId, token, "Yes!", "1", offer, _account.AuthContainer);
```

*** - secret :)
Sorry, my english is bad!

**Comment by AlekseySkynox** [comment]
Debug from RetryRequestProcessor: http://imgur.com/B3vhUI1

**Comment by FatherFoxxy** [comment]
According to the Debug, the provided AccountId and TradeOfferId were 0. The next time it crashes, please screenshot both the "Autos" and "Locals" located.

Example:
http://i.imgur.com/lIFieoe.png
http://i.imgur.com/2kxdWbp.png

**Comment by AlekseySkynox** [comment]
Thanks for the your answer! 
I'm create a four screenshots. One and two - debug from SendTradeOfferWithLink (in return). Three and four - debug from RetryRequestProcessor (in WebException).

http://imgur.com/a/uyeiF

May be problem in "CEconTradeOffer offerToken = new CEconTradeOffer { TradeOfferAccessToken = token };" ? He have one parameter, and not "AccountOtherId\TradeOfferId".

```
public SendOfferResponse SendTradeOfferWithLink(ulong partnerSid, string token, string tradeoffermessage,
            string serverid, TradeOffer offer, CookieContainer container)
        {
            const string url = "https://steamcommunity.com/tradeoffer/new/send";
            container.Add(new Cookie("bCompletedTradeOfferTutorial", "true") { Domain = "steamcommunity.com" });
            string sessionid = (from Cookie cookie in container.GetCookies(new Uri("https://steamcommunity.com"))
                                where cookie.Name == "sessionid"
                                select cookie.Value).FirstOrDefault();

            CEconTradeOffer offerToken = new CEconTradeOffer { TradeOfferAccessToken = token };

            var data = new Dictionary<string, string>
            {
                {"sessionid", sessionid},
                {"serverid", serverid},
                {"partner", partnerSid.ToString()},
                {"tradeoffermessage", tradeoffermessage},
                {"json_tradeoffer", JsonConvert.SerializeObject(offer)},
                {"captcha", string.Empty},
                {"trade_offer_create_params", JsonConvert.SerializeObject(offerToken)}
            };
            return
                _web.RetryFetch(TimeSpan.FromSeconds(10), 20, url, "POST", data, container, false,
                    $"https://steamcommunity.com/tradeoffer/new/?partner={IdConversions.UlongToAccountId(partnerSid)}&token={token}")
                    .DeserializeJson<SendOfferResponse>();
        }
```

**Comment by AlekseySkynox** [comment]
Maybe the problem is that I use the "SendTradeOfferWithLink" outside the cycle of trade?

**Comment by AlekseySkynox** [comment]
I'm use account without via a mobile application authentication. Only email. After 7 days i used SendWithToken. Maybe i'm showing all my code ?

**Comment by FatherFoxxy** [comment]
I'll try to get to you tomorrow, sorry.

**Comment by AlekseySkynox** [comment]
I'll wait! Thanks!

I'm use account without via a mobile application authentication. Only email. After 7 days i used SendWithToken. Maybe i'm showing all my code ?

**Comment by FatherFoxxy** [comment]
This may be an issue with escrow. I'll look into fixing SendTradeOffer sometime tomorrow. I don't see anything wrong with what is happening, so I assume that must be the problem.

**Comment by AlekseySkynox** [comment]
Thanks for you answer! 

I'm testing "SendTradeOfferWhithLink" with mobile protected an account. Also return an error (500). And if get trade request with "ItemsToGive" also return an error.

You my hope! :)

**Comment by FatherFoxxy** [comment]
Email confirmations are now mandatory unfortunately. SendTradeOffer 500 error was fixed in 74a4b1513d735b02419258d3caa248ba21c366d0
